gadget-container/gadgettracermanager/main.go

```diff
@@ -184,13 +185,37 @@ func main() {
 		os.Exit(1)
 	}
 
-	if dump {
+	if dump != "" {
 		out, err := client.DumpState(ctx, &pb.DumpStateRequest{})
 		if err != nil {
 			log.Fatalf("%v", err)
 		}
-		fmt.Println(out.State)
-		os.Exit(0)
+		switch dump {
+		case "":
+			// break
+
+		case "containers":
+			fmt.Println(out.Containers)
+			os.Exit(0)
+
+		case "traces":
+			fmt.Println(out.Traces)
+			os.Exit(0)
+
+		case "stacks":
+			fmt.Println(out.Stacks)
+			os.Exit(0)
+
+		case "all":
+			fmt.Println(out.Containers, out.Traces, out.Stacks)
+			os.Exit(0)
+
+		default:
+			fmt.Printf("invalid method %q\n", method)
+			flag.PrintDefaults()
+			os.Exit(1)
+		}
```

Comment:
I'm okay with either solution, but just for curiosity, why not pass what we want (containers, traces, stacks or all) in the `DumpStateRequest` and make the `DumpState` return a status (succeed or error) and a general string with the actual output?

Reply:
Thanks for the suggestion, Jose. It's a good idea and I'll keep that in mind for further approaches. Honestly, the reason why I didn't do that is because I didn't think of it. I was just happy to understand something for the first contribution 😄

Reply:
Absolutely! I think your solution works perfectly fine and I'm also happy for your first contribution 🚀 ... It is just open question because I don't know what would be the best approach and would like to have the opinion of other folks for future contributions. I don't have much experience with protobuf + gRPC and this kind of discussion is one of my main source of knowledge 🙂

Reply:
It'll be more efficient to tell the server which information the client is interested in and only return it. However, this API is for debugging purposes (not part of any "hot-path"), and additionally it's used locally through a unix socket, hence performance is not a big deal on this one.